internal/access/request.go

```diff
@@ -20,6 +22,10 @@ type RequestContext struct {
 	// start transactions. Most routes should use DBTxn and should not use
 	// DataDB directly.
 	DataDB *data.DB
+
+	// Response is a mutable field. It can be modified by API handlers to add
+	// new response metadata.
+	Response *ResponseMetadata
```

Comment:
consider making this a non-pointer

Reply:
I believe this will need to be a pointer so that we can mutate it without needing to call `c.Set()` to update the value stored in the gin.Context.  

A pointer receiver may work, but I was hoping that making this a pointer make it more obvious that this was a mutable value.  

I'll think about it some more .